std/bitmanip.d

```diff
@@ -836,11 +833,81 @@ public:
                 _ptr = b.ptr;
             }
 
+            // Set any padding bits to 0
+            if (endBits)
```

Comment:
this check is actually wrong as `newlen` should be considered rather than the current `_len`